hnn_core/tests/test_gui.py

```diff
@@ -413,3 +417,50 @@ def test_gui_adaptive_spectrogram():
                 for attr in dir(gui.viz_manager.figs[figid])]) is False
     assert len(gui.viz_manager.figs[1].axes) == 2
     plt.close('all')
+
+
+def test_unlink_relink_widget():
+    """Tests the unlinking and relinking of widgets decorator."""
+
+    # Create a basic version of the VizManager class
```

Comment:
why not use `VizManager` class directly for testing? what's the bottleneck?

Reply:
It's mostly out of principle to make the test small and clear so that the scope of the test is just the functionality of the decorator, which makes it easier to detect if the a bug is coming from within the function vs externally. 

The full functionality tests of the VizManager like`test_gui_add_figure` would already catch/debug bugs that occur in VizManager and its use of the decorator.